app/helpers/application_helper.rb

```diff
@@ -100,4 +100,22 @@ def sign_in_params(is_e2e_user:, login_type: :sign_in)
       end
     end
   end
+
+  def init_autocomplete_script(show_all_values: false, raw_attribute: false, source: false)
+    content_for(:body_end) do
+      javascript_tag defer: true do
+        "
+      document.addEventListener('DOMContentLoaded', function(event) {
+        if(window.dfeAutocomplete !== undefined && typeof window.dfeAutocomplete === 'function') {
+          dfeAutocomplete({
+            showAllValues: #{show_all_values},
+            rawAttribute: #{raw_attribute},
+            source: #{source}
+          })
+        }
+      });
+        ".html_safe
+      end
+    end
+  end
```

Comment:
This is the code views can use to initialise autocomplete components with their own options from views.

We use defer on the script tag to queue it behind our application bundle and use DOMContentLoaded to ensure it doesn't run early. I don't think it ever would given the script is at the end of the body and marked with defer.

Reply:
Neat!